inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/rocks/inspectit/oce/core/config/PropertySourcesInitializer.java

```diff
@@ -0,0 +1,46 @@
+package rocks.inspectit.oce.core.rocks.inspectit.oce.core.config;
+
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.core.env.MutablePropertySources;
+import org.springframework.core.env.PropertiesPropertySource;
+import org.springframework.core.io.ClassPathResource;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Properties;
+
+/**
+ * This class is responsible for registering all {@link org.springframework.core.env.PropertySource}s required for the initialization of inspectIT.
+ *
+ * @author Jonas Kunz
+ */
+public class PropertySourcesInitializer {
+
+    private static final String DEFAULT_CONFIG_PATH = "/config/default.yml";
+    private static final String DEFAULT_CONFIG_PROPERTYSOURCE_NAME = "inspectitDefaults";
+
+    /**
+     * Configures the {@link org.springframework.core.env.PropertySource}s of the given
+     * @param ctx the spring context
+     * @param cmdLineArgs the command line args passed to inspectIT
+     */
+    public static void configurePropertySources(AnnotationConfigApplicationContext ctx, String cmdLineArgs) {
+        MutablePropertySources propsList = ctx.getEnvironment().getPropertySources();
+
+        addCommandLineConfigDir(propsList, cmdLineArgs);
+        addAgentDefaultYaml(propsList);
+    }
+
+    private static void addAgentDefaultYaml(MutablePropertySources propsList) {
+        ClassPathResource defaultYamlResource = new ClassPathResource(DEFAULT_CONFIG_PATH, PropertySourcesInitializer.class.getClassLoader());
+        Properties defaultProps = ConfigurationUtils.readYamlsAsProperties(defaultYamlResource);
+        propsList.addLast(new PropertiesPropertySource(DEFAULT_CONFIG_PROPERTYSOURCE_NAME, defaultProps));
+    }
+
+    private static void addCommandLineConfigDir(MutablePropertySources propsList, String cmdLineArgs) {
```

Comment:
I am not convinced about this passing of the config dir with command line args..

Can we not have this somehow in the properties, so something like:
```
inspectit:
  config:
     file-config:
        enabled: false
        path: 
        watch: true
```

What I want to say is that there should be this file based config only if a user specifies it.. I guess this is again something you need to pass with the env vars in fact or? because I would like tomorrow to have

* service-config
* kubernetes-config
* etc

I am not sure if there is enough of the abstraction here at the moment.. Or you think that we need to still pass with the agruments as it could be too late? this also goes into the thingy what would we like to allow to be changed dynamically right.. 

Maybe a small discussion about this would be nice :)